README.md

```diff
@@ -1,175 +1,404 @@
+
+- **Protected SQL Execution**.
+  Work fast or safe, as needed:
+  - *Unrestricted Mode:* Provide full read/write access in development environments. Let your AI agent modify data, change the schema, drop tables, whatever you need it to do.
+  - *Restricted Mode:* Limit access in production environments with checks to ensure read-only operations and limits on resource consumption. These restrictions may become more elaborate in the future.
+
+- **Index Tuning**.
+  Ensure your SQL queries run efficiently and return quickly.
+  Find tuning targets, validate AI-generated suggestions, or generate candidates using classical index optimization algorithms.
+  Simulate how Postgres will after adding indexes using the explain plans together with [hypothetical indexes](https://hypopg.readthedocs.io/).
+
+- **Database Health**.
+  Check buffer cache hit rates, identify unused/duplicate indexes, monitor vacuum health, and more.
 
-- **Database Health Analysis**: Check buffer cache hit rates, identify unused/duplicate indexes, monitor vacuum health, and more
-- **Query Performance Analysis**: Find your slowest SQL queries and get optimization suggestions
-- **Index Recommendations**: Analyze frequently executed queries and get optimal index suggestions
-- **Schema Exploration**: Explore database tables, columns, and relationships
 
 ## Quick Start
 
-#### Using Docker
+### Prerequisites
+
+Before getting started, ensure you have:
+1. Access credentials for your database.
+2. Python 3.12 or higher *or* Docker.
+3. The `pg_statements` and `hypopg` extensions loaded on your database (for full functionality).
+
+#### Access Credentials
+ You can confirm your access credentials are valid by using `psql` or a GUI tool such as [pgAdmin](https://www.pgadmin.org/).
+
+
+#### Python  or Docker
+
+The choice to use Docker or Python is yours.
+We generally recommend using whichever is most familiar to you.
+
+#### Postgres Extensions
+
+Postgres Pro uses the `pg_statements` extension to analyze query execution statistics.
+For example, this allows it to understand which queries are running slow or consuming significant resources.
+It uses the `hypopg` extension to simulate the behavior of the Postgres query planner after adding indexes.
+While Postgres Pro can run without these extensions, its capabilities will be limited.
+
+If your Postgres database is running on a cloud provider managed service, the `pg_statements` and `hypopg` extensions will probably already be available on the system.
+In this case, you can just run `CREATE EXTENSION` commands using a role with sufficient privileges.
+
+If you are using a self-managed Postgres database, you may need to do additional work.
+Before loading the `pg_statements` extension you must ensure that it is listed in the `shared_preload_libraries` in the Postgres configuration file.
```

Comment:
(link to appendix at the bottom of the README or to another page explaining where and how to get it listed in shared_preload_libaries with an example that they can cut and paste.